core/blockchain.go

```diff
@@ -320,10 +320,9 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 		// but that's left out to a later PR since there's not really a need
 		// right now.
 		bc.stateCache.InitTransitionStatus(true, true)
-		bc.stateCache.EndVerkleTransition()
 	}
 
-	if !bc.HasState(head.Root) {
+	if !bc.stateCache.Transitioned() && !bc.HasState(head.Root) {
```

Comment:
This is an important change, in the sense that after the transition happened we can avoid checking this since the `head.Root` which is MPT related won't be found in the state and geth will go through a "state rebuilding" which will fail.

We could imagine also persisting in the database the mapping table, but that's quite complex and in the case of replays where there aren't reorgs or pruning this check isn't required. 

Note this only prevents the `if` of executing on a resumed replay. On a clean replay (starting from scratch), this will be run as usual since `Transitioned()` is `false` (and we also removed the `EndVerkleTransition()` hack mentioned previously).